nosso-calendario/src/main/java/com/br/devefc/nossocalendario/rest/UserController.java

```diff
@@ -0,0 +1,26 @@
+package com.br.devefc.nossocalendario.rest;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.br.devefc.nossocalendario.model.User;
+import com.br.devefc.nossocalendario.service.UserService;
+
+@RestController
+@RequestMapping(path = "/user")
+public class UserController {
+
+	@Autowired
+	private UserService userService;
+
+	@PostMapping(path = "/salvar")
+	public ResponseEntity<User> save(@RequestBody User user) {

```

Comment:
*Quanto mais externa a borda, mais proteção temos.*
*Usamos tudo que conhecemos que está pronto. Só fazemos código do zero se for estritamente necessário.*

Aqui você está na borda mais externa da aplicação e você **precisa** garantir que não vai entrar nada inválido na sua aplicação. O Spring já valida a entrada dos parâmetros automaticamente para você, através da Bean Validation.  Basta marcar o parâmetro com @Valid. Neste momento você está aceitando usuário com senha vazia e login vazio.

Reply:
*Separamos as bordas externas do sistema do seu núcleo. Não ligamos parâmetros de requisição externa com objetos de domínio diretamente.*

Aqui vai entrar como uma otimização prematura, mas que na minha opinião tende a ser benéfica. Neste exato momento a sua classe de domínio já possui métodos que ninguém da aplicação usa, só o Spring. Os setters e getters(esses ninguém usa mesmo por enquanto). Em algum momento o formulário fica diferente do modelo em si e aí, neste tipo de caso, você fica tentado a mudar o domínio para atender o formulário. Só que esta mesma classe de dominio já está sendo usada em outros lugares do projeto e pronto, a confusão começa :). 

Dá uma olhada nesse vídeo aqui => https://www.youtube.com/watch?v=lq5xpz79i5Q&list=PLVHlvMRWE0Y70QZLWvQvDxxQtFeOEHFkY&index=5&t=726s da playlist do meu canal, acho que pode ajudar.

Este post também ajuda => https://spring.io/guides/gs/validating-form-input/ Só não precisa do BindingResult neste momento. 

A ideia é que você o método do controller possa ser invocado independente do seu domínio, pq os parâmetros só tem conexão realmente com a web.

Reply:
Refatoração implementada, dessa forma eu consegui separar bem os domínios,  e consegui ocultar campos que não que não faziam sentido de serem exibidos na API